test/integration/DCAHubSwapper/swap-with-dex-native.spec.ts

```diff
@@ -0,0 +1,221 @@
+    USDC = await ethers.getContractAt(IERC20_ABI, USDC_ADDRESS);
+    const wethWhale = await wallet.impersonate(WETH_WHALE_ADDRESS);
+    await ethers.provider.send('hardhat_setBalance', [WETH_WHALE_ADDRESS, '0xffffffffffffffff']);
+
+    const depositAmount = RATE.mul(AMOUNT_OF_SWAPS);
+    await WETH.connect(wethWhale).transfer(cindy.address, depositAmount);
+    await WETH.connect(cindy).approve(DCAHub.address, depositAmount);
+    await DCAHub.connect(cindy)['deposit(address,address,uint256,uint32,uint32,address,(address,uint8[])[])'](
+      WETH.address,
+      USDC.address,
+      depositAmount,
+      AMOUNT_OF_SWAPS,
+      SwapInterval.ONE_MINUTE.seconds,
+      cindy.address,
+      []
+    );
+    initialPerformedSwaps = await performedSwaps();
+
+    snapshotId = await snapshot.take();
+  });
+  beforeEach('Deploy and configure', async () => {
+    await snapshot.revert(snapshotId);
+  });
+
+  describe('swap with dex', () => {
+    when('executing a swap with 0x', () => {
+      let initialHubWETHBalance: BigNumber, initialHubUSDCBalance: BigNumber, initialRecipientUSDCBalance: BigNumber;
+      let reward: BigNumber, receivedFromAgg: BigNumber;
+      given(async () => {
+        initialHubWETHBalance = await WETH.balanceOf(DCAHub.address);
+        initialHubUSDCBalance = await USDC.balanceOf(DCAHub.address);
+        initialRecipientUSDCBalance = await USDC.balanceOf(recipient.address);
+        const dexQuote = await zrx.quote({
+          chainId: 1,
+          sellToken: ETH_ADDRESS,
+          buyToken: USDC_ADDRESS,
+          sellAmount: RATE,
+          slippagePercentage: 0.01, // 1%
+          takerAddress: DCAHubSwapper.address,
+          skipValidation: true,
+        });
+        const tokensInSwap = [USDC_ADDRESS, WETH_ADDRESS];
+        const indexesInSwap = [{ indexTokenA: 0, indexTokenB: 1 }];
+        const data = encode({
+          allowanceTargets: [],
+          executions: [{ swapper: dexQuote.to, data: dexQuote.data, value: RATE }],
+          sendToProvideLeftoverToHub: true,
+        });
+        const swapTx = await DCAHubSwapper.connect(swapStarter).executeSwap(DCAHub.address, tokensInSwap, indexesInSwap, [0, 0], data, '0x', {
+          value: RATE,
+        });
+        ({ reward, receivedFromAgg } = await getTransfers(swapTx));
+      });
+      then('swap is executed', async () => {
+        expect(await performedSwaps()).to.equal(initialPerformedSwaps + 1);
+      });
+      then('hub balance is correct', async () => {
+        const hubWETHBalance = await WETH.balanceOf(DCAHub.address);
+        const hubUSDCBalance = await USDC.balanceOf(DCAHub.address);
+        expect(hubWETHBalance).to.equal(initialHubWETHBalance.sub(reward));
+        expect(hubUSDCBalance).to.equal(initialHubUSDCBalance.add(receivedFromAgg));
+      });
+      then('all reward is sent to leftover recipient', async () => {
+        const recipientWETHBalance = await WETH.balanceOf(recipient.address);
```

Comment:
See that all WETH is sent to the recipient, since we spent ETH for the swap

Reply:
Ohhhhh. It was a really delicate change. Thanks for pointing out, I would have missed it.